### DISCUSSION TITLE
hit element

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#608] [Q&A] [answered]
when i  do some markup on model,
i want know what element  hit (not select ) ?

getToolTip(hit: HitDetail) in  ToolAdmin class ,so i want how get the hit element direct from the Viewport

**Comment by bbastings** [comment, marked as answer]
Are you writing a tool? Interactive tools that want identify elements that are under the cursor can do so by setting AccuSnap.enableLocate to true in their tool's onPostInstall.

Have a look at "Writing a PrimitiveTool" in the documentation, the section on "Locate" in particular.
https://www.itwinjs.org/learning/frontend/primitivetools/

If you can provide more detail on exactly how you are doing markup I may have additional recommendations.
